Classes/Utility/ColumnLayoutUtility.php

```diff
@@ -112,4 +112,51 @@ public static function hydrateLayoutConfigFlexFormData($flexFormData): array
             }, $sheet['lDEF']);
         }, $dataStructure['data']);
     }
+
+    /**
+     * Get available and configured additional column layout classes.
+     *
+     * @param int $pageUid
+     * @return array
+     */
+    public function getAvailableLayouts(int $pageUid): array
+    {
+        $layouts = [];
+
+        // Check if the layouts are extended by ext_tables
+        if (isset($GLOBALS['TYPO3_CONF_VARS']['EXT']['column_layout']['additionalLayouts'])
+            && is_array($GLOBALS['TYPO3_CONF_VARS']['EXT']['column_layout']['additionalLayouts'])
+        ) {
+            $layouts = $GLOBALS['TYPO3_CONF_VARS']['EXT']['column_layout']['additionalLayouts'];
+        }
+
+        // Add TsConfig values
+        foreach ($this->getLayoutsFromTsConfig($pageUid) as $templateKey => $title) {
```

Comment:
Why is the variable called `$templateKey` and not `$layoutKey`? You are working with the layouts here but introducing a new concept (templates).

Reply:
I will change this. I copied some elements from tx_news and it is used for template layouts there